command/sub8_alarm/alarm_handlers/kill.py

```diff
@@ -8,13 +9,27 @@ class Kill(HandlerBase):
     def __init__(self):
         self._killed = False
         self._last_mission_killed = False
+        self.bagging_server = rospy.ServiceProxy('/online_bagger/dump', BaggerCommands)
 
     def raised(self, alarm):
         self._killed = True
+        self.bagger_dump()
 
     def cleared(self, alarm):
         self._killed = False
 
+    def bagger_dump(self):
+        """Call online_bagger/dump service"""
+        camera = '/camera/front/left/camera_info /camera/front/left/image_raw '
+        navigation = '/wrench /wrench_actual /c3_trajectory_generator/trajectory_v /c3_trajectory_generator/waypoint /trajectory '
+        controller = '/pd_out /rise_6dof/parameter_descriptions /rise_6dof/parameter_updates '
+        blueview = '/blueview_driver/ranges /blueview_driver/image_color'
+        kill_topics = camera + navigation + controller + blueview
+        try:
+            bag_status = self.bagging_server(bag_name='kill_bag', bag_time=60, topics=kill_topics)
```

Comment:
Have you tried having the sub kill unkill and kill again within a few seconds? It looks like there could be a race condition where both requests will try writing to different bags using the same bag object. That may get caught here as an exception but it'll still crash the bag server.